expression/builtin_cast_vec.go

```diff
@@ -906,11 +906,45 @@ func (b *builtinCastDecimalAsRealSig) vecEvalReal(input *chunk.Chunk, result *ch
 }
 
 func (b *builtinCastDecimalAsTimeSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDecimalAsTimeSig) vecEvalTime(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeTime(n, false)
+	result.MergeNulls(buf)
+	times := result.Times()
+	f64s := buf.Decimals()
```

Comment:
s/ f64s / decimals    is better? I think f64 is different from decimal

Reply:
Fixed